### PR TITLE
Add regression tests for cups

### PR DESCRIPTION
**Pull request by baierjan**
Enable cups regression testing during mau-extratests.

- Related ticket: https://progress.opensuse.org/issues/43133
- Verification run: 
  - http://panigale.suse.cz/tests/1050 (12-SP3)
  - http://panigale.suse.cz/tests/1052 (12-SP4)
  - http://panigale.suse.cz/tests/1051 (15)
  - http://panigale.suse.cz/tests/1053 (Leap 15.1)
  - http://panigale.suse.cz/tests/1054 (TW)

### CODE REVIEW

File: tests/console/cups.pm [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test basic capabilities of cups
+# Maintainer: Jan Baier <jbaier@suse.cz>
+
+use 5.018;
```
Comment (asmorodskyi) [inline tests/console/cups.pm]: why you need this ?

File: tests/console/cups.pm [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    record_info "access_log", "Access log should containt successful job submits";
+    assert_script_run 'grep "Send-Document successful-ok" /var/log/cups/access_log';
+
+    # Remove printers
+    record_info "lpadmin -x", "Removing printers";
+    assert_script_run "lpadmin -x $_" foreach (qw/printer_tmp printer_null/);
+    validate_script_output 'lpstat -p -d -o 2>&1 || test $? -eq 1', sub { m/No destinations added/ };
+    assert_script_run 'systemctl stop cups.service';
+    assert_script_run 'systemctl disable cups.service';
+    validate_script_output 'systemctl status cups.service || test $? -eq 3', sub { m/Active:\s*inactive/ };
```
Comment (asmorodskyi) [inline tests/console/cups.pm]: if I get you right - you ok if `systemctl disable cups.service` will return 3 , but the problem is that `assert_script_run` will fail on this so all what goes after `||`  is not reachable until you will not replace `assert_script_run` with `script_run` and check for output "manually"

File: tests/console/cups.pm [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    # Check logs
+    record_info "access_log", "Access log should containt successful job submits";
+    assert_script_run 'grep "Send-Document successful-ok" /var/log/cups/access_log';
+
+    # Remove printers
+    record_info "lpadmin -x", "Removing printers";
+    assert_script_run "lpadmin -x $_" foreach (qw/printer_tmp printer_null/);
+    validate_script_output 'lpstat -p -d -o 2>&1 || test $? -eq 1', sub { m/No destinations added/ };
+    assert_script_run 'systemctl stop cups.service';
+    assert_script_run 'systemctl disable cups.service';
```
Comment (asmorodskyi) [inline tests/console/cups.pm]: follow up on comment below :
```suggestion
    my $out = script_run 'systemctl disable cups.service';
    die "Fail to disable cups" unless $out eq 0 or $out eq 3;
```
note : this code might have "grammar" issues but you can get the idea

File: tests/console/cups.pm (hunk 2 above)
Comment (baierjan) [reply tests/console/cups.pm]: Inactive services have return code 3. So yes, the return code 3 is expected and correct.

File: tests/console/cups.pm (hunk 3 above)
Comment (baierjan) [reply tests/console/cups.pm]: I do not have problem with `assert_script_run` in this case, the problem is `validate_script_output` (which uses the assert itself).

File: tests/console/cups.pm (hunk 1 above)
Comment (baierjan) [reply tests/console/cups.pm]: That's probably copy&paste from some kernel/ltp test.

File: tests/console/cups.pm (hunk 2 above)
Comment (baierjan) [reply tests/console/cups.pm]: And one little correction, the return code 3 is from the status call and it is expected together with the "Active: inactive" line.

File: tests/console/cups.pm (hunk 1 above)
Comment (asmorodskyi) [reply tests/console/cups.pm]: please remove it

File: tests/console/cups.pm (hunk 2 above)
Comment (mimi1vx) [reply tests/console/cups.pm]: try use `systemctl` from `lib/utils.pm`

Comment (baierjan) [reply tests/console/cups.pm]: @mimi1vx If you insist.. yes, I can try.

**Comment by nilxam** [comment]
https://progress.opensuse.org/issues/45449

**Comment by DimStar77** [comment]
Since this was merged, cups test fails on TW
interestingly enough, it also failed in the published verification link given (for Leap 15.1 and TW) which begs the question why this was merged

**Pull request merged**